### PR TITLE
interaction: Implement buttons

### PR DESCRIPTION
**Pull request by chanbakjsd2**
This is currently an API in private beta so the PR is not usable unless you are part of it.

The API is subject to change but this PR should reflect a relatively new version of the API.

### CODE REVIEW

File: discord/component.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	SecondaryButton                        // Grey button.
+	SuccessButton                          // Green button.
+	DangerButton                           // Red button.
+	LinkButton                             // Button that navigates to a URL.
+)
+
+// ButtonEmoji is the emoji displayed on the button before the text.
+type ButtonEmoji struct {
+	Name     string `json:"name"`
+	ID       string `json:"id"`
```
Comment (diamondburned) [inline discord/component.go]: Is there a reason this doesn't have a specific type?

**Review by diamondburned — CHANGES_REQUESTED** [review]

File: discord/component.go (hunk 1 above)
Comment (chanbakjsd2) [reply discord/component.go]: It has been updated to be a `discord.EmojiID`.

**Review by diamondburned — DISMISSED** [review]

**Comment by diamondburned** [comment]
Holding until the official API is stable.

**Comment by yyewolf** [comment]
Aren't buttons also available on normal embeds (not using slash commands) ? (I've seen a person using them like this)

**Comment by diamondburned** [comment]
cc @chanbakjsd

**Comment by chanbakjsd2** [comment]
I'm assuming you mean buttons below embeds?
Buttons are an extra field to the reply of interactions so you can reply with the embed field and the component field to achieve that effect. The example uses `content` simply because it's the easiest (no need to construct a new embed when the point is the big struct of buttons).

**Comment by matthewpi** [comment]
https://github.com/discord/discord-api-docs/pull/3007 was just merged

**Comment by chanbakjsd2** [comment]
Unblocking because buttons API has been officially released.

**Review by diamondburned — APPROVED** [review]

**Pull request merged**